### PR TITLE
Update semconv to 1.19.0 and related build tool changes

### PR DESCRIPTION
**Pull request by lmolkova**
There were a bunch of [breaking changes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/schemas/1.19.0) in the spec in 1.19.0 along with changes in internal yaml structure and signal types.

Spec changes caused build-tool to break (https://github.com/open-telemetry/opentelemetry-specification/issues/3299) and fix requires semconv generation script changes implemented in this PR and shown in https://github.com/open-telemetry/build-tools/pull/157

### CODE REVIEW

File: semconv/src/main/java/io/opentelemetry/semconv/resource/attributes/ResourceAttributes.java [hunk 1]
```diff
@@ -666,6 +671,28 @@ public final class ResourceAttributes {
   public static final AttributeKey<String> WEBENGINE_DESCRIPTION =
       stringKey("webengine.description");
 
+  /** The name of the instrumentation scope - ({@code InstrumentationScope.Name} in OTLP). */
+  public static final AttributeKey<String> OTEL_SCOPE_NAME = stringKey("otel.scope.name");
```
Comment (lmolkova) [inline semconv/src/main/java/io/opentelemetry/semconv/resource/attributes/ResourceAttributes.java:675]: these are defined as resources https://github.com/open-telemetry/opentelemetry-specification/blob/a2a913a6dc72806749b2943ac64680cdc759dcde/semantic_conventions/scope/exporter/exporter.yaml#L2-L5, so are now generated in this file

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java [hunk 2]
```diff
@@ -36,42 +36,48 @@ public final class SemanticAttributes {
...
+   *       root</a> if there is one.
    * </ul>
    */
-  public static final AttributeKey<String> EVENT_DOMAIN = stringKey("event.domain");
-
-  /** The name of the instrumentation scope - ({@code InstrumentationScope.Name} in OTLP). */
-  public static final AttributeKey<String> OTEL_SCOPE_NAME = stringKey("otel.scope.name");
+  public static final AttributeKey<String> HTTP_ROUTE = stringKey("http.route");
 
-  /** The version of the instrumentation scope - ({@code InstrumentationScope.Version} in OTLP). */
```
Comment (lmolkova) [inline semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:57]: moved to resource attributes

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java [hunk 3]
```diff
@@ -578,30 +584,6 @@ public final class SemanticAttributes {
...
-   *       if {@code http.flavor} is {@code QUIC}, in which case {@code IP.UDP} is assumed.
-   * </ul>
-   */
-  public static final AttributeKey<String> HTTP_FLAVOR = stringKey("http.flavor");
-
-  /**
-   * Value of the <a href="https://www.rfc-editor.org/rfc/rfc9110.html#field.user-agent">HTTP
-   * User-Agent</a> header sent by the client.
-   */
-  public static final AttributeKey<String> HTTP_USER_AGENT = stringKey("http.user_agent");
```
Comment (mateuszrzeszutek) [inline semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:603]: Can you add a deprecated attribute to https://github.com/open-telemetry/opentelemetry-java/blob/main/buildscripts/semantic-convention/templates/SemanticAttributes.java.j2#L111 to minimize breaking changes?

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java [hunk 4]
```diff
@@ -315,8 +321,8 @@ public final class SemanticAttributes {
    */
   public static final AttributeKey<String> FAAS_TRIGGER = stringKey("faas.trigger");
 
-  /** The execution ID of the current function execution. */
-  public static final AttributeKey<String> FAAS_EXECUTION = stringKey("faas.execution");
```
Comment (mateuszrzeszutek) [inline semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:319]: Similar here, can you add a deprecated `FAAS_EXECUTION` pointing to `FAAS_INVOCATION_ID`?

File: semconv/src/main/java/io/opentelemetry/semconv/resource/attributes/ResourceAttributes.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply semconv/src/main/java/io/opentelemetry/semconv/resource/attributes/ResourceAttributes.java:675]: Can we leave the deprecated copies of these attributes in `SemanticAttributes` for the time being?

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java (hunk 3 above)
Comment (lmolkova) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:603]: This code is auto-generated from the spec changes and the tool does not generate removed attributes. 
I can add them manually, but they will need some handholding next time. WDYT?

Comment (marcalff) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:603]: @lmolkova There are provisions for this in the template itself.

See the code after this magic line:

```
  {%- if class == "SemanticAttributes" %}
```

For example:

* https://github.com/open-telemetry/opentelemetry-java/blob/38bf4d8ec0d794f9be77d2fe33b1ed99f5262eef/buildscripts/semantic-convention/templates/SemanticAttributes.java.j2#L121-L128

Comment (lmolkova) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:603]: @marcalff thank you for the pointer!!

**Review by jack-berg — APPROVED** [review]
I've reviewed all the keys and confirm that all are accounted for (i.e. not were erroneously deleted).

Thanks for doing this @lmolkova and for providing the fixes in `build-tools`!

**Review by mateuszrzeszutek — APPROVED** [review]
Thanks @lmolkova !

**Pull request merged**